clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -1102,10 +1102,25 @@ mlir::Value ScalarExprEmitter::VisitCastExpr(CastExpr *CE) {
     assert(E->isGLValue() && "lvalue-to-rvalue applied to r-value!");
     return Visit(const_cast<Expr *>(E));
 
-  case CK_IntegralToPointer:
-    llvm_unreachable("NYI");
-  case CK_PointerToIntegral:
-    llvm_unreachable("NYI");
+  case CK_IntegralToPointer: {
+    auto DestCIRTy = ConvertType(DestTy);
+    mlir::Value Src = Visit(const_cast<Expr *>(E));
+
+    // Properly resize by casting to an int of the same size as the pointer.
+    auto MiddleTy = CGF.CGM.getDataLayout().getIntPtrType(DestCIRTy);
+    auto MiddleVal = Builder.createIntCast(Src, MiddleTy);
+
+    if (CGF.CGM.getCodeGenOpts().StrictVTablePointers)
+      assert(UnimplementedFeature::strictVTablePointers());
```

Comment:
Do you actually need to call the unimplemented feature here? Meaning, is this path exercised on your compiler invocations? I'm currently working on a some itanium lowering and I've added asserts every time  `StrictVTablePointers` is true, which doesn't happen in my use cases. We usually add the  `UnimplementedFeature ::ABC` when it's a path we cannot assert because it will trigger a crash.

Reply:
Actually, no. I'll replace it with an `assert("NYI")`.